File: Java/Clothing.java
/*
Bismillah Saya Themy Sabri Syuhada dengan NIM 2203903.
Demi keberkahan-Nya, saya berjanji mengerjakan
latihan praktikum 2 DPBO dengan jujur dan
tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

// Clothing.java
public class Clothing extends Product {
    protected String size;
    protected String material;
    protected String gender;

    public Clothing(int idProduct, String name, String brand, double price, String size, String material,
            String gender) {
        super(idProduct, name, brand, price);
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    public String getSize() {
        return this.size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMaterial() {
        return this.material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public String getGender() {
        return this.gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String toString() {
        return super.toString() + String.format(" %-5s| %-10s| %-6s |", size, material, gender);
    }
}

File: PHP/Clothing.php
<?php

include 'Product.php';

class Clothing extends Product
{
    protected $size;
    protected $material;
    protected $gender;

    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender)
    {
        parent::__construct($idProduct, $name, $brand, $price);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    public function getSize()
    {
        return $this->size;
    }

    public function setSize($size)
    {
        $this->size = $size;
    }

    public function getMaterial()
    {
        return $this->material;
    }

    public function setMaterial($material)
    {
        $this->material = $material;
    }

    public function getGender()
    {
        return $this->gender;
    }

    public function setGender($gender)
    {
        $this->gender = $gender;
    }

    public function __toString()
    {
        return parent::__toString() . sprintf(" %-5s| %-10s| %-6s |", $this->size, $this->material, $this->gender);
    }
}
